jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -476,12 +511,18 @@ private String getServerHttpUrl(boolean useSsl) {
   private TTransport createHttpTransport() throws SQLException, TTransportException {
     CloseableHttpClient httpClient;
     boolean useSsl = isSslConnection();
-    // Create an http client from the configs
+    validateSslForBrowserMode();
     httpClient = getHttpClient(useSsl);
     transport = new THttpClient(getServerHttpUrl(useSsl), httpClient);
     return transport;
   }
 
+  protected void validateSslForBrowserMode() throws SQLException {
+    if (isBrowserAuthMode() && !isSslConnection()) {
+      throw new SQLException("Browser mode is only supported with SSL is enabled");
```

Comment:
I understand why this is done but feels odd that this throws a SQLException. Can we throw something else here and then wrap it in a SQLException in the calling method? Even a TTransportException would make sense over SQLException (though it is not thrown from thrift).